src/Api/Value/Response/ProduceResult.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace KafkaRest\Api\Value\Response;
+
+use Assert\Assertion;
+use JMS\Serializer\Annotation as Serializer;
+
+final class ProduceResult
+{
+    /**
+     * @Serializer\Type("int")
+     * @var int|null
+     */
+    public $partition;
+
+    /**
+     * @Serializer\Type("int")
+     * @var int|null
+     */
+    public $offset;
+
+    /**
+     * @Serializer\Type("int")
+     * @var int|null
+     */
+    public $errorCode;
+
+    /**
+     * @Serializer\Type("string")
+     * @var string|null
+     */
+    public $error;
+
+    public function isError() : bool
+    {
+        return $this->errorCode !== null;
+    }
+
+    public function getOffset() : Offset
+    {
+        Assertion::false($this->isError());
+
+        $offset = new Offset();
+        $offset->partition = (int) $this->partition;
+        $offset->offset = (int) $this->offset;
+
+        return $offset;
+    }
+
+    public function getError() : ProduceError
+    {
+        Assertion::true($this->isError());
+
+        $error = new ProduceError();
```

Comment:
use constructor?

Reply:
I didn't want to use constructors for objects that aren't supposed to be instantiated by a user ... does that makes sense or should I add it?

Reply:
kk